### PR TITLE
[@types/powerapps-component-framework] Update Utility type to include loadDependency

### PR DESCRIPTION
**Pull request by aroudaki**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/powerapps-component-framework/powerapps-component-framework-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6314a963a79e5a420ba3995ca2a04df423ad5c58
</p>

File: types/powerapps-component-framework/componentframework.d.ts [hunk 1]
```diff
@@ -1169,6 +1169,34 @@ declare namespace ComponentFramework {
...
+         * </resources>
+         * ```
+         * In the control code, you can load the dependency like this:
+         * ```ts
+         * context.utils.loadDependency("pubprefix_ControlNamespace.ControlName").then(() => {
+         *     // Dependent control is loaded and ready to use
+         * });
+         * ```
+         */
+        loadDependency?(dependencyName: string): Promise<unknown>;
```
Comment (pjramirez-ms) [inline types/powerapps-component-framework/componentframework.d.ts:1199]: Does the Promise return a value that can be used? If not, ```Promise<void>``` is clearer IMO.

**Review by pjramirez-ms — APPROVED** [review]

File: types/powerapps-component-framework/componentframework.d.ts (hunk 1 above)
Comment (jopursle) [reply types/powerapps-component-framework/componentframework.d.ts:1199]: Yes, it does return one of many types

**Review by jopursle — APPROVED** [review]

File: types/powerapps-component-framework/componentframework.d.ts (hunk 1 above)
Comment (aroudaki) [reply types/powerapps-component-framework/componentframework.d.ts:1199]: @jopursle any thought?

**Comment by aroudaki** [comment]
Ready to merge

**Pull request merged**